fabric-keybindings-v0/src/main/java/net/fabricmc/fabric/impl/client/keybinding/KeyBindingRegistryImpl.java

```diff
@@ -18,62 +18,40 @@
 
 import net.fabricmc.fabric.api.client.keybinding.FabricKeyBinding;
 import net.fabricmc.fabric.api.client.keybinding.KeyBindingRegistry;
+import net.fabricmc.fabric.mixin.client.keybinding.KeyBindingAccess;
 import net.minecraft.client.options.KeyBinding;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-import java.lang.reflect.Method;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 
-public class KeyBindingRegistryImpl implements KeyBindingRegistry {
+public final class KeyBindingRegistryImpl implements KeyBindingRegistry {
 	public static final KeyBindingRegistryImpl INSTANCE = new KeyBindingRegistryImpl();
 	private static final Logger LOGGER = LogManager.getLogger();
 
-	private Map<String, Integer> cachedCategoryMap;
-	private List<FabricKeyBinding> fabricKeyBindingList;
+	private final Map<String, Integer> categoryMap = KeyBindingAccess.getCategoryOrderMap();
```

Comment:
Changing this from a lazy init to being accessed indirectly through a static initializer could change behavior. We should verify that this doesn't cause any issues.